### PR TITLE
Add nsmd check for violation of excluded_prefixes

### PR DESCRIPTION
**Pull request by lobkovilya**
Signed-off-by: Ilya Lobkov <lobkovilya@yandex.ru>
## Motivation and Context

Issue https://github.com/networkservicemesh/networkservicemesh/issues/842

## How Has This Been Tested?

- [ ] Covered by existing integration testing
- [X] Added integration testing to cover
- [X] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: controlplane/pkg/nsm/nsm.go [hunk 1]
```diff
@@ -35,14 +36,14 @@ import (
...
+	DataplaneTimeout    = 15 * time.Second
 )
 
-///// Network service manager to manage both local/remote NSE connections.
+// Network service manager to manage both local/remote NSE connections.
 type networkServiceManager struct {
 	serviceRegistry  serviceregistry.ServiceRegistry
 	model            model.Model
-	excludedPrefixes []string
+	excludedPrefixes prefix_pool.PrefixPool
```
Comment (nickolaev) [inline controlplane/pkg/nsm/nsm.go]: That looks much better indeed.

File: controlplane/pkg/nsmd/constants.go [hunk 2]
```diff
@@ -1,7 +1,8 @@
 package nsmd
 
 const (
-	NsmServerSocketEnv = "NSM_SERVER_SOCKET"
-	NsmClientSocketEnv = "NSM_CLIENT_SOCKET"
-	WorkspaceEnv       = "WORKSPACE"
+	NsmServerSocketEnv  = "NSM_SERVER_SOCKET"
+	NsmClientSocketEnv  = "NSM_CLIENT_SOCKET"
+	WorkspaceEnv        = "WORKSPACE"
+	ExcludedPrefixesEnv = "EXCLUDED_PREFIXES"
```
Comment (nickolaev) [inline controlplane/pkg/nsmd/constants.go]: We should document these at some point.

File: controlplane/pkg/prefix_pool/prefixpool_test.go [hunk 3]
```diff
@@ -289,3 +289,34 @@ func TestRelease2(t *testing.T) {
 	Expect(err).To(BeNil())
 	Expect(newPrefixes).To(Equal([]string{"10.10.1.0/24"}))
 }
+
```
Comment (nickolaev) [inline controlplane/pkg/prefix_pool/prefixpool_test.go]: Testing! Great :)

File: test/integration/basic_excluded_prefixes_test.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+		t.Skip("Skip, please run without -short")
+		return
+	}
+
+	k8s, err := kube_testing.NewK8s()
+	defer k8s.Cleanup()
+	Expect(err).To(BeNil())
+
+	s1 := time.Now()
+	k8s.Prepare("nsmgr", "nsc", "nsmd-dataplane", "icmp-responder-nse")
```
Comment (nickolaev) [inline test/integration/basic_excluded_prefixes_test.go]: We use `k8s.PrepareDefault` now.

File: test/integration/basic_excluded_prefixes_test.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+	nsc, err := clientset.CoreV1().Pods("default").Create(pods.NSCPod("nsc", nodes[0].Node,
+		map[string]string{
+			"OUTGOING_NSC_LABELS": "app=icmp",
+			"OUTGOING_NSC_NAME":   "icmp-responder",
+		},
+	))
+	Expect(err).To(BeNil())
+
+	k8s.WaitLogsContains(nsc, "nsc", "srcIp intersects excludedPrefix", defaultTimeout)
```
Comment (nickolaev) [inline test/integration/basic_excluded_prefixes_test.go]: That's clever.

File: controlplane/pkg/nsmd/nsmd.go [hunk 6]
```diff
@@ -420,7 +422,30 @@ func StartAPIServerAt(server NSMServer, sock net.Listener) error {
...
+
+	excludedPrefixesEnv, ok := os.LookupEnv(ExcludedPrefixesEnv)
+	if ok {
+		logrus.Infof("Get excludedPrefixes from ENV: %v", excludedPrefixesEnv)
+		excludedPrefixes = append(excludedPrefixes, strings.Split(excludedPrefixesEnv, ",")...)
+	}
+
+	configMapPrefixes, err := getExcludedPrefixesFromConfigMap(serviceRegistry)
+	if err != nil {
+		logrus.Warnf("Cluster is not support kubeadm-config configmap, please specify PodCIDR and ServiceCIDR in %v env", ExcludedPrefixesEnv)
```
Comment (haiodo) [inline controlplane/pkg/nsmd/nsmd.go]: Should we point to documentation of howto do so?

Comment (lobkovilya) [reply controlplane/pkg/nsmd/nsmd.go]: It's temporary message, until situation with getting PodCIDR and ServiceCIDR in general case becomes clear

**Pull request merged**